recurring_contract/models/contract_group.py

```diff
@@ -304,57 +311,87 @@ def _calculate_start_date_and_offset(self):
             offset = 1
         return start_date, offset
 
-    def _should_skip_invoice_generation(self, invoicing_date):
+    def _should_skip_invoice_generation(self, invoicing_date, contract=None):
```

Comment:
@ecino Can you take the time to see if the changes here are correct ? From what I've tested it seems to work as expected but I might have missed some use cases.

I tried to not have too many side effects, so limit my changes as much as possible, this is why I check specifically for a contract from a sub proposal, which might be overkill.